java/test/src/main/java/test/Ice/idleTimeout/AllTests.java

```diff
@@ -124,6 +128,27 @@ private static void testEnableDisableIdleCheck(
         }
     }
 
+    // Verifies the idle check is disabled when the idle timeout is set to 0.
+    private static void testNoIdleTimeout(
+            String proxyString, Properties properties, PrintWriter output) {
+        output.write("testing connection with idle timeout set to 0... ");
+        output.flush();
+
+        // Create a new communicator with the desired properties.
+        properties = properties._clone();
+        properties.setProperty("Ice.Connection.Client.IdleTimeout", "0");
+        var initData = new InitializationData();
+        initData.properties = properties;
+        try (var communicator = Util.initialize(initData)) {
+            var p = TestIntfPrx.createProxy(communicator, proxyString);
+            var connection = p.ice_getConnection();
+            test(connection != null);
+            p.sleep(2000); // the implementation in the server sleeps for 2,000ms
+            connection.close();
```

Comment:
Is closing the connection necessary for the test?

Reply:
It's useful: the graceful shutdown verifies the connection was not aborted on the client-side.